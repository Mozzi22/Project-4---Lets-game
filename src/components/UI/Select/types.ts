import React from 'react';
import { IAppDiv } from 'src/components/App/styled';
import { colors } from 'src/components/UI/themeStyles';

export interface ISelect extends IAppDiv {
    id: string;
    name?: string;
    value?: string;
    onClick?: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
    onChange?: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
    content?: string;
    key?: string;
    label: string;
    margin?: string;
    padding?: string;
    width?: string;
    height?: string;
    cursor?: string;
    bgColor?: string;
    borderRadius?: string;
    border?: string;
    outline?: string;
    transition?: string;
    fontSize?: string;
    color?: string;
    focusColor?: string;
    bgColorDisabled?: string;
    options: Array<string>;
    defaultValue?: string;
    theme: 'dark' | 'light';
    colors: typeof colors;
}

export interface IOption extends ISelect {
    selected?: string;
}
