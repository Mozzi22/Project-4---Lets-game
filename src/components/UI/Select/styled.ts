import styled from 'styled-components';

import {
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  activeBorder,
} from '../baseLayout';
import { ISelect, IOption } from './types';

export const StSelect = styled.select < ISelect > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Poppins', sans-serif;
  padding: ${({ padding = '0 0 0 5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '30px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ colors, theme }) => colors[theme].colorTextForm};
  background-color: ${({ colors, theme }) => colors[theme].backgroundSelect};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius};
  border: ${({ border = activeBorder }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  option {
      background-color: ${({ colors, theme }) => colors[theme].backgroundSelect};
  }
  &:disabled{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;

export const StOption = styled.option < IOption > `
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
`;
