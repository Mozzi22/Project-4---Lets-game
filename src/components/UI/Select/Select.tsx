import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/Hooks/useTheme';
import { StSelect, StOption } from './styled';
import { ISelect } from './types';

const Select: React.FC<ISelect> = ({
    id,
    onClick,
    onChange,
    content,
    name,
    value,
    margin,
    padding,
    width,
    height,
    cursor,
    bgColor,
    borderRadius,
    border,
    outline,
    transition,
    fontSize,
    color,
    focusColor,
    bgColorDisabled,
    options,
    defaultValue,
    selected,
    }) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    return (
        <StSelect
            id={id}
            onClick={onClick}
            onChange={onChange}
            content={content}
            name={name}
            value={value}
            borderRadius={borderRadius}
            color={color}
            border={border}
            outline={outline}
            bgColorDisabled={bgColorDisabled}
            focusColor={focusColor}
            fontSize={fontSize}
            bgColor={bgColor}
            cursor={cursor}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            options={options}
            defaultValue={defaultValue}
            theme={theme}
            colors={colors}
        >
            {options.map(el => (
                <StOption
                    id={id}
                    value={el.value}
                    key={el.id}
                    onClick={onClick}
                    selected={selected}
                    bgColor={bgColor}
                >
                    {el.label}
                        {t(content)}
                </StOption>
                ))
            }
        </StSelect>
    );
};

export default Select;
