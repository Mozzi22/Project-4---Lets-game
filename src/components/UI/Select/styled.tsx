import styled from 'styled-components';
import {
  bgSelect,
  textColorBlack,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  activeBorder,
} from '../baseLayout';

import { ISelect, IOption } from './types';

export const StSelect = styled.select < ISelect > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Oxanium', cursive;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '30px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = textColorBlack }) => color};
  background-color: ${({ bgColor = bgSelect }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius};
  border: ${({ border = activeBorder }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  option {
      background-color: ${({ bgColor = bgSelect }) => bgColor};
      &:hover {
          background-color: ${({ bgColor = bgSelect }) => bgColor};
      }
  }
  &:hover {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:disabled{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;

export const StOption = styled.option < IOption > `
  font-family: 'Oxanium', cursive;
`;
