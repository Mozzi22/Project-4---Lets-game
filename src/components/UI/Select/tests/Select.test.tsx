import React from 'react';
import { shallow, mount } from 'enzyme';
import Select from '../Select';

describe('Select', () => {
  let props;
  beforeEach(() => {
    props = {
      id: 'id',
      onClick: jest.fn(),
      value: 'value',
      options: [],
    };
  });
  it('Should match snapshot', () => {
    const component = shallow(<Select {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should render Select', () => {
    const component = shallow(<Select {...props}/>);
        expect(component.find('styled__StSelect')).toHaveLength(1);
  });
  it('should call onClick', () => {
    const component = shallow(<Select {...props} />);
    component.find('styled__StSelect').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});
