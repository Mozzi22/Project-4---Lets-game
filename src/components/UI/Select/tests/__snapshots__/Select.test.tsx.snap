// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should match snapshot 1`] = `"<select id=\\"id\\" class=\\"styled__StSelect-sc-1hj7ohk-0 humWGw\\"></select>"`;
